src/Explorer/Notebook/NotebookContainerClient.ts

```diff
@@ -48,7 +49,21 @@ export class NotebookContainerClient {
     setTimeout(() => {
       this.getMemoryUsage()
         .then((memoryUsageInfo) => useNotebook.getState().setMemoryUsageInfo(memoryUsageInfo))
-        .finally(() => this.scheduleHeartbeat(Constants.Notebook.heartbeatDelayMs));
+        .catch(() => {
```

Comment:
you will have some merge conflicts with my PR on this, Changed these pieces to async-await style.  The reason for reset and allocate's scheduled jobs stepping on each other was cos this was not awaited. Please wait till my PR goes in to merge it